android/app/src/main/python/mmrelay/main.py

```
@@ -0,0 +1,422 @@
+    # Connect to Meshtastic
+    meshtastic_utils.meshtastic_client = connect_meshtastic(passed_config=config)
+
+    # Connect to Matrix
+    matrix_client = await connect_matrix(passed_config=config)
+
+    # Check if Matrix connection was successful
+    if matrix_client is None:
+        # The error is logged by connect_matrix, so we can just raise here.
+        raise ConnectionError(
+            "Failed to connect to Matrix. Cannot continue without Matrix client."
+        )
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)